dev-client/src/auth/index.ts

```diff
@@ -0,0 +1,46 @@
+import Config from 'react-native-config';
+import {OAuthProvider} from '../types';
+
+function checkFields(fields: string[], json: object) {
+  return fields.filter(key => !Object.hasOwn(json, key)).length > 0;
+}
+
+export async function exchangeToken(
+  identityJwt: string,
+  provider: OAuthProvider,
+) {
+  const resp = await fetch(Config.TERRASO_BACKEND + '/auth/token-exchange', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+    body: JSON.stringify({
+      provider,
+      jwt: identityJwt,
+    }),
+  });
```

Comment:
suggestion: the `request` method in `terraso-client-shared/terrasoApi/api` already does this request plumbing

Reply:
note you will need to provide the `Config.TERRASO_BACKEND` value in the `setApiConfig` call, since that's where `request` gets the backend URL from